app/src/main/res/layout/fragment_main_page.xml

```diff
@@ -16,67 +17,269 @@
+                        app:srcCompat="@drawable/ic_skip_next"
+                        tools:ignore="ContentDescription"
+                        />
+                    <ImageView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentLeft="true"
+                        android:layout_alignParentStart="true"
+                        android:layout_centerVertical="true"
+                        android:layout_marginLeft="@dimen/dp_10"
+                        android:layout_marginStart="@dimen/dp_10"
+                        app:srcCompat="@drawable/ic_skip_previous"
+                        tools:ignore="ContentDescription"
+                        />
+                    <!--progressbar load list shop-->
+                    <ProgressBar
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_centerHorizontal="true"
+                        android:layout_centerVertical="true"
+                        style="?android:attr/progressBarPadding"
+                        />
+                </RelativeLayout>
+
+            </android.support.design.widget.CollapsingToolbarLayout>
+        </android.support.design.widget.AppBarLayout>
+        <android.support.v4.widget.NestedScrollView
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:layout_margin="@dimen/dp_5"
+            app:layout_behavior="@string/appbar_scrolling_view_behavior"
             >
-            <TextView
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:text="@string/title_featured_product"
-                android:textColor="@color/colorAccent"
-                android:textSize="@dimen/sp_15"
-                />
-            <TextView
-                android:layout_width="wrap_content"
+            <RelativeLayout
+                android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:layout_alignParentRight="true"
-                android:onClick="@{() -> viewModel.onSeeMoreProductClick()}"
-                android:text="@string/action_see_more"
-                android:textColor="@android:color/holo_blue_dark"
-                android:textSize="@dimen/sp_15"
-                />
-        </RelativeLayout>
-        <android.support.v7.widget.RecyclerView
-            android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            app:layoutManager="@{LayoutManagers.grid(mainPage.getResources.getInteger(R.integer.item_col))}"
-            app:recyclerAdapter="@{viewModel.productAdapter}"
-            />
-    </LinearLayout>
+                >
+                <LinearLayout
```

Comment:
WARNING: [UselessParent] This `LinearLayout` layout or its `RelativeLayout` parent is useless; transfer the `background` attribute to the other view
 A layout with children that has no siblings, is not a scrollview or a root layout, and does not have a background, can be removed and have its children moved directly into the parent for a flatter and more efficient layout hierarchy.